src/session/session_api.c

```diff
@@ -166,8 +166,7 @@ __session_alter(WT_SESSION *wt_session, const char *uri, const char *config)
 	WT_WITH_CHECKPOINT_LOCK(session,
 	    WT_WITH_SCHEMA_LOCK(session,
 		ret = __wt_schema_worker(session, uri, __wt_alter, NULL, cfg,
-		WT_BTREE_ALTER |
-		WT_DHANDLE_DISCARD_FORCE | WT_DHANDLE_EXCLUSIVE)));
```

Comment:
@sueloverso, please stare at this.

The handle will be discarded regardless because `WT_BTREE_ALTER` is in `WT_BTREE_SPECIAL_FLAGS`.  I can't see any reason why mark the handle dead is preferable to ordinary (full) discard.  Please let me know if I've missed something.